### PR TITLE
Add getting started modal for commit details page

### PR DESCRIPTION
**Pull request by jeff-phillips-18**
## Fixes 
Fixes [HACBS-1012](https://issues.redhat.com/browse/HACBS-1012)

## Description

Adds a getting started info modal to the commit details view. Shown on initial visit only. Re-openable via the read more links in the tab section.

## Type of change
- [x] Feature
- [ ] Bugfix
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Screen shots / Gifs for design review 

image

image

## Browser conformance: 
- [x] Chrome
- [x] Firefox
- [x] Safari
- [x] Edge

### CODE REVIEW

**Comment by christianvogt** [comment]
designs have modal left justified and not centered.

@Ranelim @MariaLeonova do we have an updated SVG to provide @jeff-phillips-18 ?

**Comment by christianvogt** [comment]
@jeff-phillips-18 
@Ranelim had mentioned that he would like to see all modals with center justified content. Ran please confirm.

File: src/hacbs/components/Commits/CommitDetailsView.tsx [hunk 1]
```diff
@@ -140,15 +157,7 @@ const CommitDetailsView: React.FC<CommitDetailsViewProps> = ({ commitName, appli
...
-                Component:{' '}
-                {commit.components.map((component, index) => {
-                  return (
-                    <>
-                      {component}
-                      {index < commit.components.length - 1 && ','}
-                    </>
-                  );
-                })}
+                Component:{`${commit.components.join(', ')}`}
```
Comment (christianvogt) [inline src/hacbs/components/Commits/CommitDetailsView.tsx:160]:
```suggestion
                Component: {`${commit.components.join(', ')}`}
```

File: src/hacbs/components/Commits/CommitsGettingStartedModal.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  onHide: () => void;
+};
+
+const CommitsGettingStartedModal: React.FC<CommitsGettingStartedModalProps> = ({
+  shown,
+  onHide,
+}) => {
+  const header = (
+    <Title className="pf-u-text-align-center" headingLevel="h2">
+      Getting started with Commit view
```
Comment (christianvogt) [inline src/hacbs/components/Commits/CommitsGettingStartedModal.tsx:16]:
```suggestion
      Getting started with commits view
```

File: src/hacbs/components/Commits/CommitsGettingStartedModal.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  const body = (
+    <>
+      <div className="pf-u-display-flex pf-u-justify-content-center pf-u-flex-fill">
+        <img src={imageUrl} alt="getting started with commits" />
+      </div>
+      <Text className="pf-u-text-align-left pf-u-mt-md">
+        View Commit activity, triggered pipelines and environment deployment in a single view.
+        <br />
+        Each node represents a github action, pipeline or an environment.
+      </Text>
```
Comment (christianvogt) [inline src/hacbs/components/Commits/CommitsGettingStartedModal.tsx:29]: Not that we have this interaction right now, but it's in the design.
```suggestion
        <br />
        Click on nodes to view more details.
      </Text>
```

File: src/hacbs/components/Commits/tabs/CommitsOverviewTab.tsx [hunk 4]
```diff
@@ -1,13 +1,28 @@
...
+  <PageSection padding={{ default: 'noPadding' }} variant={PageSectionVariants.light} isFilled>
+    <Text className="pf-u-mt-lg pf-u-mb-lg">
+      Events progression triggered by the commit.
+      <Button
+        data-testid="hacbs-commit-overview-learn-more"
+        variant="link"
+        className="pf-u-pl-sm"
+        onClick={onLearnMore}
+      >
+        read more
```
Comment (christianvogt) [inline src/hacbs/components/Commits/tabs/CommitsOverviewTab.tsx:21]: For consistency with other pages (cc @Ranelim)
```suggestion
        Learn more
```

File: src/hacbs/components/Commits/CommitsGettingStartedModal.tsx (hunk 2 above)
Comment (jeff-phillips-18) [reply src/hacbs/components/Commits/CommitsGettingStartedModal.tsx:16]: Maybe `Getting started with the Commit view`? The view is for a single commit.

Comment (christianvogt) [reply src/hacbs/components/Commits/CommitsGettingStartedModal.tsx:16]: @Ranelim @MariaLeonova please advise.

But really it's not that big of a deal as we'll be sweeping text later when microcopy is updated.

**Comment by jeff-phillips-18** [comment]
Updated image:
image

**Comment by Ranelim** [comment]
@jeff-phillips-18 
Looks cool, Thank you!
The modal width should match the image.

I'm not sure this is related but wanted to mention it since there's an image for it here:
I see also the commit page itself has some stuff we want to fix, like placing the GitHub icon and the status label next to the page header. It was something we postponed but I think @christianvogt said we can do it now.

Thank you!

**Comment by christianvogt** [comment]
@Ranelim the other things you mentioned aside from the modal width are beyond the scope of this PR

**Comment by jeff-phillips-18** [comment]
image

**Pull request merged**